packages/consensus/src/codec.cairo

```diff
@@ -28,54 +29,69 @@ pub impl EncodeSpan<T, +Encode<T>> of Encode<Span<T>> {
 }
 
 pub impl EncodeByteArray of Encode<ByteArray> {
-    fn encode_to(self: @ByteArray, ref dest: ByteArray) {
+    fn encode_to(self: @ByteArray, ref dest: Array<u32>) {
         encode_compact_size(self.len(), ref dest);
-        dest.append(self);
+        for i in 0..self.len() {
+            dest.append(self[i].into());
```

Comment:
@maciejka this will work ok probably only for `P2TR` and `P2SH` scripts, which length is divisible for 4
Other script types would introduce shifts and we won't be able to use this encoding.